### PR TITLE
[moe] support moe fwd and bwd with low level zero

### PR DESCRIPTION
**Pull request by oahzxl**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [ ] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: tests/test_moe/test_moe_checkpoint.py [hunk 1]
```diff
@@ -11,7 +11,17 @@
 from colossalai.utils import get_current_device
 from colossalai.zero import ColoInitContext
 from tests.test_moe.test_moe_zero_init import MoeModel
-from tests.test_zero.test_legacy.common import CONFIG
+
+CONFIG = dict(fp16=dict(mode=None,),
+              zero=dict(level=3,
```
Comment (ver217) [inline tests/test_moe/test_moe_checkpoint.py:16]: Can we remove the dependency of config?

File: tests/test_moe/test_moe_zero_init.py [hunk 2]
```diff
@@ -50,45 +50,17 @@ def forward(self, x):
...
-        else:
-            assert param.colo_attr.data_payload.device.type == 'cuda'
+def run_moe_zero_init():
+    model = MoeModel(checkpoint=True)
+    plugin = LowLevelZeroPlugin(initial_scale=2**5)
+    booster = Booster(plugin=plugin)
+    model = booster.boost(model)[0]
+
+    # assert local expert number
+    assert len(model.module.test_transform.moe.moe_layer.experts.experts) == 8 // MOE_CONTEXT.world_size
```
Comment (ver217) [inline tests/test_moe/test_moe_zero_init.py:60]: This check can be done in fwd_bwd check. Doing so can reduce time of launch multiprocess.

File: tests/test_moe/test_moe_checkpoint.py (hunk 1 above)
Comment (oahzxl) [reply tests/test_moe/test_moe_checkpoint.py]: ok, i have moved config

File: tests/test_moe/test_moe_zero_init.py (hunk 2 above)
Comment (oahzxl) [reply tests/test_moe/test_moe_zero_init.py]: ok, i move it to the fwd bwd check

**Review by ver217 — APPROVED** [review]

**Pull request merged**